.github/workflows/build.yml

```diff
@@ -0,0 +1,39 @@
+name: Build
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+    paths-ignore:
+      - '**.rst'
+      - 'docs/**'
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**.rst'
+      - 'docs/**'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+  
+jobs:
+  build_wheels:
+    strategy:
+      matrix:
+        os: ["ubuntu-20.04", "macos-11"]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install dependencies
+      run: |
+        make requirements c_lib
+    - name: Build wheels
+      uses: pypa/cibuildwheel@v2.12.3
+    - name: Upload packages
+      uses: actions/upload-artifact@v3
```

Comment:
Do we want to upload packages for pull_requests that are opened?

Reply:
I went with "yes" so I can download the produced wheels and test them for compatibility.

Note that this is "uploading" to the GHA zip file and not to a package index.

Reply:
This can be easily abused by people. Can you limit it to `main` branch only?

Reply:
I changed the workflow file to exclude pull requests. You can see that [this run](https://github.com/ada-url/ada-python/actions/runs/5028036512) did not upload artifacts.